src/http/HttpHeaderBag.php

```diff
@@ -37,28 +59,97 @@ public function get($key) {
     if (array_key_exists($key, $this->values)) {
       return $this->values[$key];
     }
-    return NULL;
+    // Return a NULL object on which you can still call the HttpHeaderInterface
+    // methods.
+    return HttpHeaderNull::create(NULL, NULL);
```

Comment:
Or maybe you could have already gotten the string/ array from the ``get()`` - ``function get($string = TRUE)``

```php
$string = $headers->get('header-that-does-not-exist');
$values = $headers->get('header-that-does-not-exist', FALSE);
```